pkg/relayer/proxy/synchronous.go

```diff
@@ -82,18 +85,44 @@ func (sync *synchronousRPCServer) Stop(ctx context.Context) error {
 	return sync.server.Shutdown(ctx)
 }
 
-// Service returns the underlying service object.
-func (sync *synchronousRPCServer) Service() *sharedtypes.Service {
-	return sync.service
-}
-
 // ServeHTTP listens for incoming relay requests. It implements the respective
 // method of the http.Handler interface. It is called by http.ListenAndServe()
 // when synchronousRPCServer is used as an http.Handler with an http.Server.
 // (see https://pkg.go.dev/net/http#Handler)
 func (sync *synchronousRPCServer) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 	ctx := request.Context()
 
+	var originHost string
+	// Get the original host from X-Forwarded-Host header if specified in the proxy config.
```

Comment:
Why do we ned to do this?

Reply:
Updated the comments to explain why we have this.